clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CacheContainerResourceDefinition.java

```diff
@@ -207,6 +209,16 @@ public ModelNode transformOperation(ModelNode operation) {
                 }
             };
             builder.addRawOperationTransformationOverride(ListOperations.LIST_REMOVE_DEFINITION.getName(), new SimpleOperationTransformer(removeAliasTransformer));
+
+            // Reject if using embedded thread pool configuration
+            builder.rejectChildResource(ThreadPoolResourceDefinition.WILDCARD_PATH);
+        } else {
+            for (ThreadPoolResourceDefinition pool : ThreadPoolResourceDefinition.values()) {
+                pool.buildTransformation(version, parent);
+            }
+            for (ScheduledThreadPoolResourceDefinition pool : ScheduledThreadPoolResourceDefinition.values()) {
+                pool.buildTransformation(version, parent);
+            }
```

Comment:
I think this might be the reason why this PR broke support for legacy slaves.  Shouldn't this transformation logic be triggered based on InfinispanModel.VERSION_4_0 instead of InfinispanModel.VERSION_3_0?